OSDOCS-9882 OCP 4.16 Dev Console Bug Fix RNs (#77695)
openshift/openshift-docs: release_notes/ocp-4-16-release-notes.adoc

[id="ocp-4-16-dev-console-bug-fixes_{context}"]
==== Developer Console

> * Previously, when a serverless function was created in the create serverless form, BuilldConfig was not created. With this update, if the Pipelines Operator is not installed, if the pipeline resource is not created for particular resource, or if the pipeline is not added while creating a serverless function, it will create BuildConfig as expected. (link:https://issues.redhat.com/browse/OCPBUGS-34143[*OCPBUGS-34143*])

Reviewer: ```suggestion
* Previously, when a serverless function was created in the create serverless form, `BuilldConfig` was not created. With this update, if the Pipelines Operator is not installed, the pipeline resource is not created for particular resource, or the pipeline is not added while creating a serverless function, it will create `BuildConfig` as expected. (link:https://issues.redhat.com/browse/OCPBUGS-34143[*OCPBUGS-34143*])
```

> * Previously, after installing the Pipelines Operator, Pipeline templates took some time to become available in the cluster, but users were still able to create the Deployment. With this update, the *Create* button on the *Import from Git* page is be disabled if there is no pipeline template present for the resource selected. (link:https://issues.redhat.com/browse/OCPBUGS-34142[*OCPBUGS-34142*])

Reviewer: ```suggestion
* Previously, after installing the Pipelines Operator, Pipeline templates took some time to become available in the cluster, but users were still able to create the deployment. With this update, the *Create* button on the *Import from Git* page is disabled if there is no pipeline template present for the resource selected. (link:https://issues.redhat.com/browse/OCPBUGS-34142[*OCPBUGS-34142*])
```

> * Previously, the maximum number of nodes was set to 100 on the *topology* page. A persistant alert, "Loading is taking longer than expected." was provided. With this update, the limit of nodes is increased to 300. (link:https://issues.redhat.com/browse/OCPBUGS-32307[*OCPBUGS-32307*])

Reviewer: ```suggestion
* Previously, the maximum number of nodes was set to `100` on the *Topology* page. A persistant alert, "Loading is taking longer than expected." was provided. With this update, the limit of nodes is increased to `300`. (link:https://issues.redhat.com/browse/OCPBUGS-32307[*OCPBUGS-32307*])
```

> * With this update, an alert message to notify you that Service Bindings are deprecated with {product-titile} 4.15 was added to the *ServiceBinding list*, *ServiceBinding details*, *Add*, and *Topology* pages when creating a ServiceBinding, binding a component, or if a ServiceBinding was found in the current namespace. (link:https://issues.redhat.com/browse/OCPBUGS-32222[*OCPBUGS-32222*])

Reviewer: ```suggestion
* With this update, an alert message to notify you that Service Bindings are deprecated with {product-title} 4.15 was added to the *ServiceBinding list*, *ServiceBinding details*, *Add*, and *Topology* pages when creating a `ServiceBinding`, binding a component, or a `ServiceBinding` was found in the current namespace. (link:https://issues.redhat.com/browse/OCPBUGS-32222[*OCPBUGS-32222*])
```

* Previously, the Helm Plugin index view did not display the same number of charts as the Helm CLI if the chart names varied. With this release, the Helm catalog now looks for `charts.openshift.io/name` and `charts.openshift.io/provider` so that all versions are grouped together in a single catalog title. (link:https://issues.redhat.com/browse/OCPBUGS-32059[*OCPBUGS-32059*])

* Previously, the `TaskRun` status was not displayed near the `TaskRun` name on the *TaskRun details* page. With this update, the `TaskRun` status is located beside the name of the `TaskRun` in the page heading. (link:https://issues.redhat.com/browse/OCPBUGS-31745[*OCPBUGS-31745*])

* Previously, there is an error when adding parameters to the Pipeline when the resources field was added to the payload, and as resources are deprecated. With this update, the resources fields have been removed from the payload, and you can add parameters to the Pipeline. (link:https://issues.redhat.com/browse/OCPBUGS-31082[*OCPBUGS-31082*])

> * This release updates the pipelines plugin to support the latest Pipeline Trigger API version for the custom resource definitions (CRDs) `ClusterTriggerBinding`, `TriggerTemplate` and `EventListener`. (link:https://issues.redhat.com/browse/OCPBUGS-30958[*OCPBUGS-30958*])

Reviewer: ```suggestion
* This release updates the OpenShift Pipelines plugin to support the latest Pipeline Trigger API version for the custom resource definitions (CRDs) `ClusterTriggerBinding`, `TriggerTemplate` and `EventListener`. (link:https://issues.redhat.com/browse/OCPBUGS-30958[*OCPBUGS-30958*])
```

* Previously, `CustomTasks` were not recognized or remained in a `Pending` state. With this update, `CustomTasks` can be easily identified from the Pipelines *List* and *Details* pages. (link:https://issues.redhat.com/browse/OCPBUGS-29513[*OCPBUGS-29513*])

> * Previously, if there was a build output image with an `Image` tag, then the `Output Image` link, which was not redirecting to the correct *ImageStream* page. With this update, this has been fixed by generating a URL for the *ImageStream* page without adding the tag in the link. (link:https://issues.redhat.com/browse/OCPBUGS-29355[*OCPBUGS-29355*])

Reviewer: ```suggestion
* Previously, if there was a build output image with an `Image` tag then the `Output Image` link would not redirect to the correct *ImageStream* page. With this update, this has been fixed by generating a URL for the *ImageStream* page without adding the tag in the link. (link:https://issues.redhat.com/browse/OCPBUGS-29355[*OCPBUGS-29355*])
```

* Previously, `BuildRun` logs were not visible in the *Logs* tab of the *BuildRun* page due to a recent update in the API version of the specified resources. With this update, the logs of the `TaskRuns` were added back into the *Logs* tab of the `BuildRun` page for both v1alpha1 and v1beta1 versions of the Builds Operator. (link:https://issues.redhat.com/browse/OCPBUGS-27473[*OCPBUGS-27473*])

* Previously, the annotations to set scale bound values were setting to `autoscaling.knative.dev/maxScale` and `autoscaling.knative.dev/minScale`. With this update, the annotations to set scale bound values are updated to `autoscaling.knative.dev/min-scale` and `autoscaling.knative.dev/max-scale` to determine the minimum and maximum numbers of replicas that can serve an application at any given time. You can set scale bounds for an application to help prevent cold starts or control computing costs. (link:https://issues.redhat.com/browse/OCPBUGS-27469[*OCPBUGS-27469*])

> * Previously, the *Log* tab for *PipelineRuns* from the Tekton Results API never finishes loading. With this release, this tab loads correct for PipelineRuns loaded from the Kubernetes API or the Tekton Results API. (link:https://issues.redhat.com/browse/OCPBUGS-25612[*OCPBUGS-25612*])

Reviewer: ```suggestion
* Previously, the *Log* tab for *PipelineRuns* from the Tekton Results API never finished loading. With this release, this tab loads fully complete for PipelineRuns loaded from the Kubernetes API or the Tekton Results API. (link:https://issues.redhat.com/browse/OCPBUGS-25612[*OCPBUGS-25612*])
```
